R/log_formatter.R

```diff
@@ -0,0 +1,34 @@
+#' Teal `log_formatter`
+#'
+#' Custom `log_formatter` supporting atomic vectors. By default `glue::glue`
+#' returns n-element vector when vector is provided as an input. This function
+#' generates `"[elem1, elem2, ...]"` for atomic vectors. Function also handles
+#' `NULL` value which normally causes `logger` to return empty character.
+#' @name teal_logger_formatter
+#' @return (`character(1)`) formatted log entry
+#' @keywords internal
+teal_logger_formatter <- function() {
+  logger::log_formatter(
+    function(..., .logcall = sys.call(), .topcall = sys.call(-1), .topenv = parent.frame()) {
+      logger::formatter_glue(
+        ...,
+        .null = "NULL", .logcall = .logcall, .topcall = .topcall, .topenv = .topenv,
+        .transformer = teal_logger_transformer
+      )
+    }
+  )
+}
+
+#' @rdname teal_logger_formatter
+#' @inheritParams glue::identity_transformer
+teal_logger_transformer <- function(text, envir) {
+  value <- glue::identity_transformer(text, envir)
+  if (is.null(value)) {
+    "NULL"
+  } else if (is.vector(value)) {
+    utils::capture.output(expr <- dput(value)) # dput prints
+    deparse1(expr)
+  } else {
+    glue::glue_collapse(value, sep = ", ")
```

Comment:
maybe also with a `width` parameter set to a super high value in case of hitting some edge case here

Reply:
I find `glue_collapse` not needed at all. Theoretically we can always `dput(value)` which can determine call/literal for any object `NULL`, `character(0)`, `vectors`, `matrices`, `lists`. I think `glue` is primarily thought to be used in similar way to `sprintf` - when object has n-elements then it output n-elements. In logs we use `glue` but our purpose is to concat the object so that it fits to a single character-output.

Reply:
great - even better 👍 
aren't we afraid about very lenghty string? what if value is a long vector with lenghty strings?

Reply:
`deparse1` has a default limit 500 characters. In general, I think if somebody doesn't like a produced log, if it is too long or something, then one can handle it when calling `logger::log_`. What we guarantee now is:
- handling any R object type passed to glue-`{ }` 
- 500-chars limit of each "glued" item. 

Theoretically my change didn't really affected a length of the logs - default `glue_collapse(..., width = Inf)`, so it still prints lengthy objects.